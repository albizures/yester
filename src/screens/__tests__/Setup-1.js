import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import Setup1 from '../Setup-1'

const navigation = { navigate: jest.fn() }

describe('src/screens/Setup-1.js', () => {
  it('should render', () => {
    const tree = renderer.create(<Setup1 navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to Setup 2 is pressed', () => {
    it('should navigate to Setup2 screen', () => {
      const shallowSetup1 = shallow(<Setup1 navigation={navigation} />)
      const shallowTouchableHighlights = shallowSetup1.find(Button)
      const shallowButtonToHome = shallowTouchableHighlights.at(0)

      shallowButtonToHome.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Setup2')
    })
  })
})
