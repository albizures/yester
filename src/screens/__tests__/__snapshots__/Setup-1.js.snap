// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/Setup-1.js should render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    SETUP-1
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        to Setup-2
      </Text>
    </View>
  </View>
</View>
`;
